crates/bevy_reflect/src/type_name.rs

```diff
@@ -0,0 +1,66 @@
+use crate::utility::GenericTypeNameCell;
+
+/// Provide the name of the type as string.
```

Comment:
Nit: Could you give some more information here? Maybe why it's useful or what it's needed for?